crypto/zk/circuit/circuit.go

```diff
@@ -211,7 +211,13 @@ func downloadFile(ctx context.Context, fileUrl string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer res.Body.Close()
+
+	defer func() {
+		if err := res.Body.Close(); err != nil {
+			log.Warnf("error closing body response %v", err)
+		}
+	}()
+
```

Comment:
we're just reading from the http body, so I'm pretty sure we can safely ignore the close error. if the download started and then failed for any reason, `io.ReadAll` would catch it.

Reply:
as a general rule of thumb, `Close` methods should have their errors checked if you're writing to the file/object, since in some cases the "close" is what triggers a flush or some last checks. When reading, `Close` is just about releasing resources, because you're not actually sending any data that you need to ensure got received/stored properly.

Reply:
Thanks! @mvdan